awscli/compat.py

```diff
@@ -87,6 +114,8 @@ def compat_open(filename, mode='r', encoding=None):
 
     raw_input = raw_input
 
+    binary_stdin = sys.stdin
```

Comment:
I guess for me, while I think trying to combine all of these is a good idea to reduce technical debt as it is strange that we have two different ways we print binary to standard out. I think I would prefer to minimize the amount of moving parts in this change as we have never relied on changing the mode of stdout or stdin on windows before with streaming s3 commands. I want to avoid as many regressions as possible since this change is quite a big change already.

Can we settle on adding a big todo note on combining/unifying these? I plan once ``s3transfer`` is in place to go through and just get rid of anything that may be technical debt because I have a feeling there will be a fair amount of technical debt left and the s3 codebase is already due for a cleanup.